In [27]:
# udemy_courses.csv
import pandas as pd
df=pd.read_csv("udemy_courses.csv")
# if it is a excel file
# df=pd.read_excel('dataset.xlsx')

# if data in in tab seperated fashion 
# df=pd.read_csv('dataset.txt',delimiter='\t')

df.head(2)
# to fetch first two rows of the dataset/dataframe

course_id                                       course_title  \
0    1070968                 Ultimate Investment Banking Course   
1    1113822  Complete GST Course & Certification - Grow You...   

                                                 url  is_paid  price  \
0  https://www.udemy.com/ultimate-investment-bank...     True    200   
1      https://www.udemy.com/goods-and-services-tax/     True     75   

   num_subscribers  num_reviews  num_lectures       level  content_duration  \
0             2147           23            51  All Levels               1.5   
1             2792          923           274  All Levels              39.0   

    published_timestamp           subject  
0  2017-01-18T20:58:58Z  Business Finance  
1  2017-03-09T16:34:20Z  Business Finance

In [28]:
# to get list of columns
print(df.columns)

Index(['course_id', 'course_title', 'url', 'is_paid', 'price',
       'num_subscribers', 'num_reviews', 'num_lectures', 'level',
       'content_duration', 'published_timestamp', 'subject'],
      dtype='object')


In [29]:
# The dataset contains 4 numeric feilds [course_id , price , num_subscribers , num_reviews , num_lectures , content_duration]
# describe() is a function that give aggregate values of each numeric feild/featue/column
df.describe()

course_id        price  num_subscribers   num_reviews  num_lectures  \
count  3.678000e+03  3678.000000      3678.000000   3678.000000   3678.000000   
mean   6.759720e+05    66.049483      3197.150625    156.259108     40.108755   
std    3.432732e+05    61.005755      9504.117010    935.452044     50.383346   
min    8.324000e+03     0.000000         0.000000      0.000000      0.000000   
25%    4.076925e+05    20.000000       111.000000      4.000000     15.000000   
50%    6.879170e+05    45.000000       911.500000     18.000000     25.000000   
75%    9.613555e+05    95.000000      2546.000000     67.000000     45.750000   
max    1.282064e+06   200.000000    268923.000000  27445.000000    779.000000   

       content_duration  
count       3678.000000  
mean           4.094517  
std            6.053840  
min            0.000000  
25%            1.000000  
50%            2.000000  
75%            4.500000  
max           78.500000

In [30]:
# check for what type of data we have    📄📄
# info gives information of column and there data types ,  null values , memory occupied and dimensions  
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3678 entries, 0 to 3677
Data columns (total 12 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   course_id            3678 non-null   int64  
 1   course_title         3678 non-null   object 
 2   url                  3678 non-null   object 
 3   is_paid              3678 non-null   bool   
 4   price                3678 non-null   int64  
 5   num_subscribers      3678 non-null   int64  
 6   num_reviews          3678 non-null   int64  
 7   num_lectures         3678 non-null   int64  
 8   level                3678 non-null   object 
 9   content_duration     3678 non-null   float64
 10  published_timestamp  3678 non-null   object 
 11  subject              3678 non-null   object 
dtypes: bool(1), float64(1), int64(5), object(5)
memory usage: 319.8+ KB


## Findings from above two function   🕵️🔍

1. The highest price of single course is 200.00
2. The lowest price of single course is 0.00
3. The average price of course is 66.04
4. 78.5 is the longest course in udemy 
5. 4.09 is the average lenght of course on udemy
6. There are 3678 courses on udemy and there are no empty cell in any column
7. There are 12 columns in the dataset

In [31]:
type(df)

pandas.core.frame.DataFrame

### Access data with respect to column by names

In [32]:
# df['course_title']                        
df[['course_title','price']]

course_title  price
0                    Ultimate Investment Banking Course    200
1     Complete GST Course & Certification - Grow You...     75
2     Financial Modeling for Business Analysts and C...     45
3     Beginner to Pro - Financial Analysis in Excel ...     95
4          How To Maximize Your Profits Trading Options    200
...                                                 ...    ...
3673  Learn jQuery from Scratch - Master of JavaScri...    100
3674  How To Design A WordPress Website With No Codi...     25
3675                      Learn and Build using Polymer     40
3676  CSS Animations: Create Amazing Effects on Your...     50
3677  Using MODX CMS to Build Websites: A Beginner's...     45

[3678 rows x 2 columns]

### Access rows with index 

In [33]:
df.iloc[10:13]   # the psitioning of rows is based on indexing
# if i want 10,11,12 rows

# df.iloc[:10]
# if i want first 10 rows

#  for index, row in df.iterrows():
#     print(index, row['Name'])
#   'x' stores index on the row and 'y' i think this will be in dictonary for of the row

course_id                                       course_title  \
10     592338  Forex Trading Secrets of the Pros With Amazon'...   
11     975046                    Trading Options With Money Flow   
12     742602  Financial Management Risk and Return For Secur...   

                                                  url  is_paid  price  \
10  https://www.udemy.com/trading-with-amazons-aws...     True    200   
11  https://www.udemy.com/trading-options-using-mo...     True    200   
12     https://www.udemy.com/risk-and-return-finance/     True     30   

    num_subscribers  num_reviews  num_lectures       level  content_duration  \
10             4284           93            76  All Levels               5.0   
11             1380           42            17  All Levels               1.0   
12             3607           21            19  All Levels               1.5   

     published_timestamp           subject  
10  2015-09-11T16:47:02Z  Business Finance  
11  2016-10-18T22:52:31Z  Business Finance  
12  2016-02-03T18:04:01Z  Business Finance

In [34]:
print(df.iloc[2,1])
# to access the 1'st column of 2 row 

Financial Modeling for Business Analysts and Consultants


## Rearrange columns
#### remove colums that you don't want

In [37]:
col=df.columns
df=df[['course_id','course_title','is_paid','subject','price','content_duration','level','num_lectures','url']]
df.head(1)

course_id                        course_title  is_paid           subject  \
0    1070968  Ultimate Investment Banking Course     True  Business Finance   

   price  content_duration       level  num_lectures  \
0    200               1.5  All Levels            51   

                                                 url  
0  https://www.udemy.com/ultimate-investment-bank...

###  Sort the rows based on any specific column data

In [38]:
#df.sort_values('course_id')   # default descinding order
df.sort_values('course_id',ascending=False)

course_id                                       course_title  is_paid  \
455     1282064  Cryptocurrency (BTC & ETH) Investment & Tradin...     True   
2645    1277924                            Angular 4 Front To Back     True   
638     1276364        Bitcoin & Ethereum Course (2 Course Bundle)     True   
461     1276182  Options Trading Foundation: Your journey to co...     True   
1264    1276020                Photoshop para iniciante - o básico     True   
...         ...                                                ...      ...   
2700      11475                   Become a Certified Web Developer     True   
2701      11174                Become a Web Developer from Scratch     True   
3082      11153     Learn jQuery: An In-depth Course For Beginners     True   
2491       8325            HTML Tutorial: HTML & CSS for Beginners     True   
2985       8324                           Javascript for Beginners     True   

               subject  price  content_duration               level  \
455   Business Finance     20               2.5      Beginner Level   
2645   Web Development     40               8.5          All Levels   
638   Business Finance    200               2.5          All Levels   
461   Business Finance     95               1.0  Intermediate Level   
1264    Graphic Design     30               3.0      Beginner Level   
...                ...    ...               ...                 ...   
2700   Web Development    100              12.0          All Levels   
2701   Web Development    120              27.5          All Levels   
3082   Web Development     50               4.5          All Levels   
2491   Web Development     20               4.0          All Levels   
2985   Web Development     20               3.0          All Levels   

      num_lectures                                                url  
455              7  https://www.udemy.com/cryptocurrency-btc-eth-i...  
2645            68     https://www.udemy.com/angular-4-front-to-back/  
638             52     https://www.udemy.com/bitcoin-ethereum-bundle/  
461              5  https://www.udemy.com/option-trading-foundatio...  
1264            20  https://www.udemy.com/photoshop-para-iniciante...  
...            ...                                                ...  
2700           185  https://www.udemy.com/become-a-certified-web-d...  
2701           197  https://www.udemy.com/how-to-become-a-web-deve...  
3082            62                      https://www.udemy.com/jquery/  
2491            82                 https://www.udemy.com/learn-html5/  
2985            48        https://www.udemy.com/beginning-javascript/  

[3678 rows x 9 columns]

### sort rows based on price and then duration

In [43]:
df.sort_values(['price','content_duration'],ascending=[False,True])[['course_title','price','content_duration']]

course_title  price  \
587   1 - Concepts of Statistics For Beginners Step ...    200   
538                           Intro to Impact Investing    200   
1307  Make tech circles in Adobe Illustrator in 30 m...    200   
415   Coaching Course:Investment Analysis for your c...    200   
802          Trade the News in Forex: EA Robot Included    200   
...                                                 ...    ...   
2827               Learn HTML5 Programming From Scratch      0   
2649  Learn E-Commerce Website in PHP & MySQL From S...      0   
589   الدروس التعليمية لسوق الفوركس والأسواق المالية...      0   
3513           Foundations of Front-End Web Development      0   
3336  Learn Bootstrap3 With Multi Purposes Single Pa...      0   

      content_duration  
587           0.516667  
538           0.533333  
1307          0.533333  
415           0.566667  
802           0.566667  
...                ...  
2827         10.500000  
2649         17.500000  
589          20.500000  
3513         20.500000  
3336         21.000000  

[3678 rows x 3 columns]

In [47]:
df['price'].sum()
# df['content_duration'].sum()
# sum of prices and duration of all courses

15059.633333333335

###  To filter rows with condition

In [65]:
filter=df.loc[(df['price']>=200) & (df['subject']=='Business Finance')&(df['num_lectures']>165)]
filter

course_id                                       course_title  is_paid  \
44      301442       Black Algo Trading: Build Your Trading Robot     True   
160     543242                            Learn to Trade The News     True   
500     521342  Succeed in Forex Even if You Don't Know Where ...     True   

              subject  price  content_duration           level  num_lectures  \
44   Business Finance    200              16.0      All Levels           227   
160  Business Finance    200              46.5      All Levels           544   
500  Business Finance    200              19.5  Beginner Level           211   

                                                   url  
44     https://www.udemy.com/build-your-trading-robot/  
160     https://www.udemy.com/learn-to-trade-the-news/  
500  https://www.udemy.com/forex-trading-for-beginn...

In [66]:
filter.reset_index(drop=True,inplace=True)
filter

course_id                                       course_title  is_paid  \
0     301442       Black Algo Trading: Build Your Trading Robot     True   
1     543242                            Learn to Trade The News     True   
2     521342  Succeed in Forex Even if You Don't Know Where ...     True   

            subject  price  content_duration           level  num_lectures  \
0  Business Finance    200              16.0      All Levels           227   
1  Business Finance    200              46.5      All Levels           544   
2  Business Finance    200              19.5  Beginner Level           211   

                                                 url  
0    https://www.udemy.com/build-your-trading-robot/  
1     https://www.udemy.com/learn-to-trade-the-news/  
2  https://www.udemy.com/forex-trading-for-beginn...

In [72]:

df.groupby(['subject', 'level']).count()['price']


subject              level             
Business Finance     All Levels            696
                     Beginner Level        340
                     Expert Level           31
                     Intermediate Level    128
Graphic Design       All Levels            298
                     Beginner Level        243
                     Expert Level            5
                     Intermediate Level     57
Musical Instruments  All Levels            276
                     Beginner Level        296
                     Expert Level            7
                     Intermediate Level    101
Web Development      All Levels            659
                     Beginner Level        391
                     Expert Level           15
                     Intermediate Level    135
Name: price, dtype: int64

### findings from the above code
1. Here we can see how many All Levels, Beginner Level ,Expert Level , Intermediate Level   to a specific subject